enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+annotations included in the allowed list - others may also be considered.
+
+### Risks and Mitigations
+
+**Risk:** Sensitive information can be leaked to unauthorized users.
+
+_Mitigations:_ The proposal for the Projected-resource CSI driver includes RBAC protections and
+SELinux controls to ensure secrets are protected.
+
+**Risk:** Allowing arbitrary annotations in builds can lead to privilege escalations.
+
+_Mitigation:_ Builds will use an allowlist to block arbitrary annotations from being passed through
+to build pods.
+
+## Design Details
+
+### Test Plan
+
+N/A - each individual component will have their own test plan.
+
+### Graduation Criteria
+
+The following is a proposed release timeline:
+
+1. The Projected-resource CSI driver needs to reach maturity via a Tech Preview -> GA cycle.
+2. The subscription content webhook will explicitly depend on the Projected-resource CSI driver.
+   It’s lifecycle is tied down wrt Tech Preview -> GA Builds will require 2 and 3 in order to pass
+   subscription content through.
+3. The cluster subscription manager can be released as an independent component on its own cadence.
+   However, it should employ the conventions of the subscription content webhook to ensure seamless
+   behavior.
+
+### Upgrade / Downgrade Strategy
+
+Generally speaking, each of the components in this meta-enhancement will be add-ons, deliverable via
+OLM. That said, a case can be made that these should be added to the core OpenShift payload.
+
+### Version Skew Strategy
+
+Each component will be responsible for managing version skew during upgrade.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Drawbacks
+
+Consuming subscription content was a core feature of OCP 3.x, and was provided by subscribing every
+node in the OpenShift cluster. If these components are delivered via OLM, then consuming
+subscription becomes an “extra” feature.
+
+The projected resource driver also introduces a potential attack vector. Without proper precautions,
+an attacker can gain access to sensitive information.
+
+## Alternatives
+
+As a work-around, customers can create a MachineConfig that adds the same entitlement keys to
+/etc/pki/entitlement/ on every node [1]. CRI-O and buildah are configured by default to mount these
+entitlement keys into all running containers.
+
+Instead of using a projected resource storage driver, we could create a mechanism where entitlements
+are added to the cluster, and then all MachineConfigs/MachineSets are updated accordingly. This has
+a downside of forcing all nodes to be restarted when subscriptions are added or updated. The current
+proposal does not have this drawback.
+
+[1] https://access.redhat.com/solutions/4908771
+
+## Infrastructure Needed [optional]
+
+- New Github repos will be needed for each component above.
+- New CI templates needed to install the respective components for e2e testing.
```

Comment:
And one or more new CVO Operators would be needed?

Reply:
Indicating this will be an OLM operator